app/controllers/application_controller.rb

```diff
@@ -1,10 +1,142 @@
 require './config/environment'
+require 'pry'
 
 class ApplicationController < Sinatra::Base
+  register Sinatra::ActiveRecordExtension
 
   configure do
     set :public_folder, 'public'
     set :views, 'app/views'
+    enable :sessions
+    set :session_secret, "secret"
   end
 
+  get '/' do
+  	erb :index
+  end
+
+  get '/tweets' do
+  	if logged_in?
+  		erb :'/tweets/tweets'	
+ 	else
+	  	redirect '/login'
+  	end
+  end
+
+  get '/tweets/new' do
+  	if logged_in?
+	  	erb :'/tweets/create_tweet'
+	 else
+	  	redirect '/login'
+	 end
+  end
+
+  post '/tweets' do
+  	if params[:content] != ""
+	  	@tweet = Tweet.new(content: params[:content], user_id: session[:id])
+	  	@tweet.save
+	  	redirect "/tweets/#{@tweet.id}"
+	else
+		redirect "/tweets/new"
+	end
+  end
+
+  get '/tweets/:id' do
+  	if logged_in?
+	  	@tweet = Tweet.find_by_id(params[:id])
+	  	erb :'/tweets/show_tweet'
+	 else
+	  	redirect '/login'
+  	end
+  end
+
+  get '/tweets/:id/edit' do
+  	if logged_in?
+	  	@user = current_user
+	  	@tweet = Tweet.find_by_id(params[:id])
+	  	erb :'tweets/edit_tweet'
+	  else
+	  	redirect '/login'
+  	end
+  end
+
+  patch '/tweets/:id' do
+  	@tweet = Tweet.find_by_id(params[:id])
+  	if params[:content] != "" 
+	  	@tweet.update(content: params[:content])
+	  	@tweet.save
+	  	redirect "/tweets/#{@tweet.id}"
+  	else
+  		redirect "/tweets/#{@tweet.id}/edit"
+  	end
+
+  end
+
+  post '/tweets/:id/delete' do 
+  	@tweet = Tweet.find_by_id(params[:id])
+  	if @tweet.user_id == current_user.id
```

Comment:
Prefer to find tweets that are already only associated to the `current_user` than to manually check the `user_id` foreign key of the tweet against the `current_user.id`

```ruby
tweet = current_user.tweets.find_by(id: params[:id])
if tweet && tweet.destroy
  redirect "/tweets"
else
  redirect "/tweets/#{tweet.id}"
end
```